src/main/kotlin/ai/serenade/intellij/services/CommandHandler.kt

```diff
@@ -162,6 +162,15 @@ class CommandHandler(private val project: Project) {
     /*
      * Wrappers
      */
+
+    private fun checkModality(action: () -> CallbackData?): CallbackData? {
```

Comment:
Ah didn't realize this wasn't applicable to the other cases (error reporting done via EditorState and redo/undo infinite loop was caused by something else). Both this and prior logic both make sense.